python/lsst/meas/astrom/pessimistic_pattern_matcher_b_3D.py

```diff
@@ -0,0 +1,722 @@
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+from scipy.spatial import cKDTree
+
+
+class PessimisticPatternMatcherB(object):
+    """Class implimenting a pessimistic version of Optimsitic Pattern Matcher B
+    from Tabur 2007. The class loads and stores the reference object in a
+    convienent data structure for matching any set of source obejcts that are
+    assumed to contain each other. The pessimistic nature of the algorithm
+    comes from requiring that it discovers at least two patterns that agree on
+    the correct shift and rotation for matching before exiting. The original
+    behavior of OPMb can be recovered simply.
```

Comment:
Changed all "function" descriptions as described. Added a short description of what a pattern to be matched looks like and referred the interested party to the current DMTN that I am currently writing.